### DISCUSSION TITLE
[Support] No internet access after install

### DISCUSSION THREAD

**Discussion by GEUchiha** [pivpn/pivpn#1407] [Support]
## In raising this issue, I confirm the following:

`{please fill the checkboxes, e.g: [X]}`

- [x] I have read the [documentation](https://docs.pivpn.io)
- [x] I have read and understood the [contributors guide](https://github.com/pivpn/pivpn/blob/master/CONTRIBUTING.md).
- [x] The issue I am reporting can be *replicated*.
- [] The issue I am reporting can be *is* directly related to the pivpn installer script.

### Have you searched for similar issues and solutions?
    yes
https://forums.raspberrypi.com/viewtopic.php?t=241234
https://github.com/pivpn/pivpn/issues/1309
https://github.com/pivpn/pivpn/issues/774
https://github.com/pivpn/pivpn/issues/752
https://github.com/pivpn/pivpn/issues/207
https://github.com/pivpn/pivpn/issues/182
https://github.com/pivpn/pivpn/issues/151
https://gist.github.com/laurenorsini/9925434
https://github.com/pivpn/pivpn/issues/634
https://forums.raspberrypi.com/viewtopic.php?t=265457

### Has your install failed? 
```
no
```

#### What type of hardware are you running PiVPN at?

```
Raspberrypi (4B)

```
#### Output of `uname -a`
```
Linux raspberrypi 5.10.63-v7l+ #1459 SMP Wed Oct 6 16:41:57 BST 2021 armv7l GNU/Linux

```

#### Output of `cat /etc/os-release`
```
PRETTY_NAME="Raspbian GNU/Linux 10 (buster)"
NAME="Raspbian GNU/Linux"
VERSION_ID="10"
VERSION="10 (buster)"
VERSION_CODENAME=buster
ID=raspbian
ID_LIKE=debian
HOME_URL="http://www.raspbian.org/"
SUPPORT_URL="http://www.raspbian.org/RaspbianForums"
BUG_REPORT_URL="http://www.raspbian.org/RaspbianBugs"

```

### Issue
After connect to VPN through my iphone XS MAX ios 14.1 with openvpn.
i was able to connect to my local device's IP address (such as my openmediavault control page).
I was able to receive and send out message through WeChat app.
But **not able**:
browser other app or get on to other webpage. only WeChat worked for few minutes. After that I am not able to receive or send out any thing (seems no access to internet only lan)

I do not have pihole, only installed openmediavault and aria2.
Did Preliminary checks, and **Packet capture did not have respond when i try to connect something from my iphone (mobile data)**

### Console output of      `curl -L install.pivpn.io | bash`
```
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100    17  100    17    0     0     22      0 --:--:-- --:--:-- --:--:--    22
100   162  100   162    0     0     97      0  0:00:01  0:00:01 --:--:--   230
100 82115  100 82115    0     0  42002      0  0:00:01  0:00:01 --:--:--  169k
:::
::: sudo will be used for the install.
::: Cancel selected. Exiting

```

### Console output of      `pivpn add` or `pivpn add nopass`
```
I was able to create user and get .ovpn file without issue
```

### Console output of      `pivpn debug`

```
::: Generating Debug Output
::::            PiVPN debug              ::::
=============================================
::::            Latest commit            ::::
Branch: master
Commit: 0883893a4fdce436213ffb0bf514c94f5224722a
Author: 4s3ti
Date: Mon Nov 1 22:02:25 2021 +0100
Summary: Merge branch 'test'
=============================================
::::        Installation settings        ::::
PLAT=Raspbian
OSCN=buster
USING_UFW=0
IPv4dev=eth0
IPv4addr=192.168.1.103/24
IPv4gw=192.168.1.1
install_user=pi
install_home=/home/pi
VPN=openvpn
pivpnPROTO=udp
pivpnPORT=1194
pivpnDNS1=8.8.8.8
pivpnDNS2=8.8.4.4
pivpnSEARCHDOMAIN=
pivpnHOST=REDACTED
TWO_POINT_FOUR=1
pivpnENCRYPT=256
USE_PREDEFINED_DH_PARAM=
INPUT_CHAIN_EDITED=1
FORWARD_CHAIN_EDITED=0
pivpnDEV=tun0
pivpnNET=10.8.0.0
subnetClass=24
ALLOWED_IPS=""
UNATTUPG=1
INSTALLED_PACKAGES=(openvpn grepcidr expect unattended-upgrades)
HELP_SHOWN=1
=============================================
::::  Server configuration shown below   ::::
dev tun
proto udp
port 1194
ca /etc/openvpn/easy-rsa/pki/ca.crt
cert /etc/openvpn/easy-rsa/pki/issued/raspberrypi_ab7bc4f4-ec5c-4e66-81ac-740abeb220d4.crt
key /etc/openvpn/easy-rsa/pki/private/raspberrypi_ab7bc4f4-ec5c-4e66-81ac-740abeb220d4.key
dh none
ecdh-curve prime256v1
topology subnet
server 10.8.0.0 255.255.255.0
# Set your primary domain name server address for clients
push "dhcp-option DNS 8.8.8.8"
push "dhcp-option DNS 8.8.4.4"
# Prevent DNS leaks on Windows
push "block-outside-dns"
# Override the Client default gateway by using 0.0.0.0/1 and
# 128.0.0.0/1 rather than 0.0.0.0/0. This has the benefit of
# overriding but not wiping out the original default gateway.
push "redirect-gateway def1"
client-to-client
client-config-dir /etc/openvpn/ccd
keepalive 15 120
remote-cert-tls client
tls-version-min 1.2
tls-crypt /etc/openvpn/easy-rsa/pki/ta.key
cipher AES-256-CBC
auth SHA256
user openvpn
group openvpn
persist-key
persist-tun
crl-verify /etc/openvpn/crl.pem
status /var/log/openvpn-status.log 20
status-version 3
syslog
verb 3
#DuplicateCNs allow access control on a less-granular, per user basis.
#Remove # if you will manage access by user instead of device.
#duplicate-cn
# Generated for use by PiVPN.io
=============================================
::::  Client template file shown below   ::::
client
dev tun
proto udp
remote REDACTED 1194
resolv-retry infinite
nobind
remote-cert-tls server
tls-version-min 1.2
verify-x509-name raspberrypi_ab7bc4f4-ec5c-4e66-81ac-740abeb220d4 name
cipher AES-256-CBC
auth SHA256
auth-nocache
verb 3
=============================================
::::    Recursive list of files in       ::::
::: /etc/openvpn/easy-rsa/pki shows below :::
/etc/openvpn/easy-rsa/pki/:
Default.txt
ca.crt
crl.pem
ecparams
huang_family.ovpn
index.txt
index.txt.attr
index.txt.attr.old
index.txt.old
issued
openssl-easyrsa.cnf
private
renewed
revoked
safessl-easyrsa.cnf
serial
serial.old
ta.key

/etc/openvpn/easy-rsa/pki/ecparams:
prime256v1.pem

/etc/openvpn/easy-rsa/pki/issued:
huang_family.crt
raspberrypi_ab7bc4f4-ec5c-4e66-81ac-740abeb220d4.crt

/etc/openvpn/easy-rsa/pki/private:
ca.key
huang_family.key
raspberrypi_ab7bc4f4-ec5c-4e66-81ac-740abeb220d4.key

/etc/openvpn/easy-rsa/pki/renewed:
private_by_serial
reqs_by_serial

/etc/openvpn/easy-rsa/pki/renewed/private_by_serial:

/etc/openvpn/easy-rsa/pki/renewed/reqs_by_serial:

/etc/openvpn/easy-rsa/pki/revoked:
private_by_serial
reqs_by_serial

/etc/openvpn/easy-rsa/pki/revoked/private_by_serial:

/etc/openvpn/easy-rsa/pki/revoked/reqs_by_serial:
=============================================
::::            Self check               ::::
:: [OK] IP forwarding is enabled
:: [OK] Iptables MASQUERADE rule set
:: [OK] Iptables INPUT rule set
:: [OK] OpenVPN is running
:: [OK] OpenVPN is enabled (it will automatically start on reboot)
:: [OK] OpenVPN is listening on port 1194/udp
=============================================
:::: Having trouble connecting? Take a look at the FAQ:
:::: https://docs.pivpn.io/faq
=============================================
::::      Snippet of the server log      ::::
Nov 15 11:33:26 raspberrypi ovpn-server[2499]: REDACTED:9186 VERIFY OK: depth=0, CN=huang_family
Nov 15 11:33:26 raspberrypi ovpn-server[2499]: REDACTED:9186 peer info: IV_VER=3.git::58b92569
Nov 15 11:33:26 raspberrypi ovpn-server[2499]: REDACTED:9186 peer info: IV_PLAT=ios
Nov 15 11:33:26 raspberrypi ovpn-server[2499]: REDACTED:9186 peer info: IV_NCP=2
Nov 15 11:33:26 raspberrypi ovpn-server[2499]: REDACTED:9186 peer info: IV_TCPNL=1
Nov 15 11:33:26 raspberrypi ovpn-server[2499]: REDACTED:9186 peer info: IV_PROTO=2
Nov 15 11:33:26 raspberrypi ovpn-server[2499]: REDACTED:9186 peer info: IV_AUTO_SESS=1
Nov 15 11:33:26 raspberrypi ovpn-server[2499]: REDACTED:9186 peer info: IV_GUI_VER=net.openvpn.connect.ios_3.2.3-3760
Nov 15 11:33:26 raspberrypi ovpn-server[2499]: REDACTED:9186 peer info: IV_SSO=openurl
Nov 15 11:33:26 raspberrypi ovpn-server[2499]: REDACTED:9186 Control Channel: TLSv1.3, cipher TLSv1.3 TLS_AES_256_GCM_SHA384, 256 bit EC, curve: prime256v1
Nov 15 11:33:26 raspberrypi ovpn-server[2499]: REDACTED:9186 [huang_family] Peer Connection Initiated with [AF_INET]REDACTED:9186
Nov 15 11:33:26 raspberrypi ovpn-server[2499]: huang_family/REDACTED:9186 OPTIONS IMPORT: reading client specific options from: /etc/openvpn/ccd/huang_family
Nov 15 11:33:26 raspberrypi ovpn-server[2499]: huang_family/REDACTED:9186 MULTI: Learn: 10.8.0.2 -> huang_family/REDACTED:9186
Nov 15 11:33:26 raspberrypi ovpn-server[2499]: huang_family/REDACTED:9186 MULTI: primary virtual IP for huang_family/REDACTED:9186: 10.8.0.2
Nov 15 11:33:26 raspberrypi ovpn-server[2499]: huang_family/REDACTED:9186 PUSH: Received control message: 'PUSH_REQUEST'
Nov 15 11:33:26 raspberrypi ovpn-server[2499]: huang_family/REDACTED:9186 SENT CONTROL [huang_family]: 'PUSH_REPLY,dhcp-option DNS 8.8.8.8,dhcp-option DNS 8.8.4.4,block-outside-dns,redirect-gateway def1,route-gateway 10.8.0.1,topology subnet,ping 15,ping-restart 120,ifconfig 10.8.0.2 255.255.255.0,peer-id 0,cipher AES-256-GCM' (status=1)
Nov 15 11:33:26 raspberrypi ovpn-server[2499]: huang_family/REDACTED:9186 Data Channel: using negotiated cipher 'AES-256-GCM'
Nov 15 11:33:26 raspberrypi ovpn-server[2499]: huang_family/REDACTED:9186 Outgoing Data Channel: Cipher 'AES-256-GCM' initialized with 256 bit key
Nov 15 11:33:26 raspberrypi ovpn-server[2499]: huang_family/REDACTED:9186 Incoming Data Channel: Cipher 'AES-256-GCM' initialized with 256 bit key
Nov 15 11:33:44 raspberrypi ovpn-server[2499]: huang_family/REDACTED:9186 SIGTERM[soft,remote-exit] received, client-instance exiting
=============================================
::::            Debug complete           ::::
:::
::: Debug output completed above.
::: Copy saved to /tmp/debug.log
:::

```
### Have you taken any steps towards solving your issue?
```

Other than the steps from F&Q, i've also tried editing server.conf（https://gist.github.com/laurenorsini/9925434）and steps from the posts i've searched.
I've also tried re-install pivpn.

```

**Comment by coolapso** [comment]
Hi, 

Thanks for using PiVPN, and for making such a great support request, always nice to see when people fill everything properly and provide proper data! 

I have moved this to discussions, 
Why it have worked in the beginning and then stopped I cannot understand, but it all looks like your VPN has you behind a NAT, do you know this? can you double check it?

**Reply by GEUchiha** [reply]
Hello, its also great to have such nice tool and active detailed support.

I have very limited knowledge about setting up the network; but I've searched lots of guide and found/followed the guide that most suits to my internet environment.

Please find my overview of pi internet setting include the install pivpn steps, **hope this can answer 'your VPN has you behind a NAT'** question due to im not sure how to check:

- PI connected to the wall (which connected to my modern provide by ISP)
- i've gotten static external IP from ISP, and installed PPPOE on PI, and got the IP address after setup pppoe
- installed PIVPN
   [1st DHCP setting] i've set the local ip address (192.168.1.103) and gateway (192.168.1.1)
   [2nd static ip setting] PIVPN setup page auto recognized my static external IP
   [3rd add user] Added user and send .ovpn file to my phone with mobile data
- **Last step, i've login to ISP's modern, and did port forwarding, see img below**
https://i.imgur.com/8g6bWWc.png

I think you are asking if i've set up port forwarding? not sure if the image can answers or not. I am not sure what it can mean.

### [Additional step i did after your question]
I've tried disable my port forwarding from the last step, and ran ```sudo systemctl restart openvpn```

### [Result]
In openVPN, i still only have 0 ~ 200B/S, and 0~ 400B/S out.
Basically its still the same. The weird thing the first time connection is another chat app (this time was "Line" - line is the chatapp) received 1 or 2 messages. But after i open the chat app, nothing comes in or out. Reconnecting it won't make any different.



However, i am very positive about the first time success when Wechat was able to send out 3 messages while VPN connected. I am still trying to use ```sudo tcpdump -n -i eth0 udp port 1194``` to see i can get anything or not, but still no luck.


are there other possible cause of this? or perhaps i can run more other commands and get you more details.

**Reply by TinCanTech** [reply]
According to the snippet of the server log, your client is connected normally:
```
Nov 15 11:33:26 raspberrypi ovpn-server[2499]: huang_family/REDACTED:9186 PUSH: Received control message: 'PUSH_REQUEST'
Nov 15 11:33:26 raspberrypi ovpn-server[2499]: huang_family/REDACTED:9186 SENT CONTROL [huang_family]: 'PUSH_REPLY,dhcp-option DNS 8.8.8.8,dhcp-option DNS 8.8.4.4,block-outside-dns,redirect-gateway def1,route-gateway 10.8.0.1,topology subnet,ping 15,ping-restart 120,ifconfig 10.8.0.2 255.255.255.0,peer-id 0,cipher AES-256-GCM' (status=1)
Nov 15 11:33:26 raspberrypi ovpn-server[2499]: huang_family/REDACTED:9186 Data Channel: using negotiated cipher 'AES-256-GCM'
Nov 15 11:33:26 raspberrypi ovpn-server[2499]: huang_family/REDACTED:9186 Outgoing Data Channel: Cipher 'AES-256-GCM' initialized with 256 bit key
Nov 15 11:33:26 raspberrypi ovpn-server[2499]: huang_family/REDACTED:9186 Incoming Data Channel: Cipher 'AES-256-GCM' initialized with 256 bit key
```
Does the client connect to the public IP ? (In client.conf does `remote` point to a public IP not a private IP)

**Reply by GEUchiha** [reply]
> According to the snippet of the server log, your client is connected normally:
> 
> ```
> Nov 15 11:33:26 raspberrypi ovpn-server[2499]: huang_family/REDACTED:9186 PUSH: Received control message: 'PUSH_REQUEST'
> Nov 15 11:33:26 raspberrypi ovpn-server[2499]: huang_family/REDACTED:9186 SENT CONTROL [huang_family]: 'PUSH_REPLY,dhcp-option DNS 8.8.8.8,dhcp-option DNS 8.8.4.4,block-outside-dns,redirect-gateway def1,route-gateway 10.8.0.1,topology subnet,ping 15,ping-restart 120,ifconfig 10.8.0.2 255.255.255.0,peer-id 0,cipher AES-256-GCM' (status=1)
> Nov 15 11:33:26 raspberrypi ovpn-server[2499]: huang_family/REDACTED:9186 Data Channel: using negotiated cipher 'AES-256-GCM'
> Nov 15 11:33:26 raspberrypi ovpn-server[2499]: huang_family/REDACTED:9186 Outgoing Data Channel: Cipher 'AES-256-GCM' initialized with 256 bit key
> Nov 15 11:33:26 raspberrypi ovpn-server[2499]: huang_family/REDACTED:9186 Incoming Data Channel: Cipher 'AES-256-GCM' initialized with 256 bit key
> ```
> 
> Does the client connect to the public IP ? (In client.conf does `remote` point to a public IP not a private IP)

Thanks for checking.

From the client side (IOS OPENVPN), i have not find 'client.conf' file, but below is my info after connect
[YOUR PRIVTAE IP]
10.8.0.2

[SERVER]
218.11.45.131 (made this up, this IP is the static public IP which i got from ISP, and it is the IP i have on my PI after PPPOE setup

[SERVER PUBLIC IP]
218.11.45.131, it is same as IP address.

From I understand is, the client connected to 218.11.45.131 server (which is my PI), and got assigned a virtual IP which is 10.8.0.2 (also LAN to other device). Since it got assigned a virtual IP, it should also have internet access ability from the server (218.11.45.131)

[additional info connection log from openvpn]
https://i.imgur.com/hOzaQNQ.png

**Reply by TinCanTech** [reply]
Your connection looks normal.  Try testing DNS by `ping 8.8.8.8` vs `ping google.com`

**Reply by GEUchiha** [reply]
> Your connection looks normal. Try testing DNS by `ping 8.8.8.8` vs `ping google.com`

I've also tried some other DNS, and results are the same. Below is the ping for google DNS:

**Pinging 8.8.8.8 with 32 bytes of data:**
Reply from 8.8.8.8: bytes=32 time=10ms TTL=118
Reply from 8.8.8.8: bytes=32 time=10ms TTL=118
Reply from 8.8.8.8: bytes=32 time=11ms TTL=118
Reply from 8.8.8.8: bytes=32 time=12ms TTL=118

Ping statistics for 8.8.8.8:
    Packets: Sent = 4, Received = 4, Lost = 0 (0% loss),
Approximate round trip times in milli-seconds:
    Minimum = 10ms, Maximum = 12ms, Average = 10ms



**Pinging google.com [172.217.160.110] with 32 bytes of data:**
Reply from 172.217.160.110: bytes=32 time=15ms TTL=118
Reply from 172.217.160.110: bytes=32 time=29ms TTL=118
Reply from 172.217.160.110: bytes=32 time=13ms TTL=118
Reply from 172.217.160.110: bytes=32 time=92ms TTL=118

Ping statistics for 172.217.160.110:
    Packets: Sent = 4, Received = 4, Lost = 0 (0% loss),
Approximate round trip times in milli-seconds:
    Minimum = 13ms, Maximum = 92ms, Average = 37ms

I've just got 2 message on another chatting app while VPN was on, but not able to get on all other access. I had _TCPDUMP_ on, and it did not show up any packets.

**Reply by TinCanTech** [reply]
Then you were not using `tcpdump` correctly:
```
tcpdump -ni eth0
```
Change `eth0` to the correct name.

**Reply by GEUchiha** [reply]
`tcpdump -n -i eth0 udp port 1194`
```
pi@raspberrypi:~ $ sudo tcpdump -n -i eth0 udp port 1194
tcpdump: verbose output suppressed, use -v or -vv for full protocol decode
listening on eth0, link-type EN10MB (Ethernet), capture size 262144 bytes
```
I've used above code from F&Q, and it got me nothing when 2 message received from my device. Now i think the message i receive might be just some leftover data... im not so sure



`tcpdump -ni eth0`

When i use this, i got lots of data

```
pi@raspberrypi:~ $ sudo tcpdump -ni eth0
tcpdump: verbose output suppressed, use -v or -vv for full protocol decode
listening on eth0, link-type EN10MB (Ethernet), capture size 262144 bytes
17:03:40.935009 PPPoE  [ses 0x34d3] IP 218.11.45.131.22 > 60.209.70.189.42879: Flags [P.], seq 2371722818:2371723006, ack 2620258163, win 501, length 188
17:03:40.935812 PPPoE  [ses 0x34d3] IP 218.11.45.131.22 > 60.209.70.189.42879: Flags [P.], seq 188:384, ack 1, win 501, length 196
17:03:40.936117 PPPoE  [ses 0x34d3] IP 218.11.45.131.22 > 60.209.70.189.42879: Flags [P.], seq 384:556, ack 1, win 501, length 172
17:03:40.936410 PPPoE  [ses 0x34d3] IP 218.11.45.131.22 > 60.209.70.189.42879: Flags [P.], seq 556:728, ack 1, win 501, length 172

```
`
218.11.45.131 is my static external IP
60.209.70.189 is some IP i don't recognize 
above arent my real ip
`
When i have VPN connected on my iphone with mobile data, i only see 2 IPs from the log (218.11.45.131 and 60.209.70.189). Either one IP from the left, or one IP from the right.


Anyway, since the log you've seen that i've setup all correctly, just might be internet structure issue (ISP)? Any suggestion that i can continue dig into this issue?
